### PR TITLE
importing raise_error and molutil

### PR DESCRIPTION
**Pull request by loriab**
- [x] I'm not wholly convinced, but add `raise_error` for default to `which` and `which_import` after discussion in https://github.com/MolSSI/QCEngine/pull/65
- [x] Replace deprecated `find_loader` with `find_spec`.
- [x] Q: `ModuleNotFoundError` is _not_ right for `which` but what is?
- [x] Move numpy alignment to molutil

### CODE REVIEW

File: qcelemental/tests/test_importing.py [hunk 1]
```diff
@@ -23,6 +25,13 @@ def test_which_import_f_bool():
     assert ans is False
 
 
+def test_which_import_f_raise():
+    with pytest.raises(ModuleNotFoundError) as e:
+        qcel.util.which_import('evilpint', raise_error=True)
+
+    assert "'evilpint' not found" in str(e)
```
Comment (dgasmith) [inline qcelemental/tests/test_importing.py]: Hmm... no such thing.

File: qcelemental/util/importing.py [hunk 2]
```diff
@@ -40,12 +48,20 @@ def which(command, *, return_bool=False):
...
+
     """
-    lenv = {'PATH': ':' + os.environ.get('PATH') + ":" + os.path.dirname(sys.executable)}
+    lenv = {'PATH': ':' + os.environ.get('PATH', '') + ':' + os.path.dirname(sys.executable)}
     lenv = {k: v for k, v in lenv.items() if v is not None}
 
     ans = shutil.which(command, mode=os.F_OK | os.X_OK, path=lenv['PATH'])
 
+    if raise_error and ans is None:
+        raise ModuleNotFoundError(f"Command '{command}' not found in envvar PATH.")
```
Comment (dgasmith) [inline qcelemental/util/importing.py]: I think you convinced me of this one where we should always add custom messages. What if we added:

`"Command '{command}' not found in envvar PATH. {if install else ''}"`

or similar here? Would hide a bit more logic.

Comment (loriab) [reply qcelemental/util/importing.py]: Where `install` is a message passed into the fn? I decided against that once, but not strenuously. Fine by me.

**Comment by loriab** [comment]
this is RTG

**Comment by dgasmith** [comment]
Squish LGTM commits?

**Comment by loriab** [comment]
squish the whole thing, imo

**Comment by loriab** [comment]
Fixed some `os.pathsep` and added an `env` to `which` to help code with custom envvar. Not sure I did it the best way. Here's a context manager: https://github.com/laurent-laporte-pro/stackoverflow-q2059482/blob/master/demo/environ_ctx.py

**Comment by dgasmith** [comment]
LGTM

**Pull request merged**